### PR TITLE
add index to table

### PR DESCRIPTION
**Pull request by empresarrollo**
Just added an index to the scheduled_notifications table, so the query is more optimized when you have lots of notications created.

### CODE REVIEW

File: migrations/2019_11_13_175222_add_index_to_scheduled_notifications_table.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table(config('snooze.table'), function (Blueprint $table) {
+            $table->dropIndex(['send_at']);
+        });
+    }
+}
```
Comment (atymic) [inline migrations/2019_11_13_175222_add_index_to_scheduled_notifications_table.php]: newline please

**Comment by atymic** [comment]
Good idea!

I'd prefer to add it to the existing migration however, could you update that?

Thanks

**Comment by empresarrollo** [comment]
> I'd prefer to add it to the existing migration however, could you update that?

Are you sure? I think the current migrations should not be touched, moreso in a package. Because you then break the migrations for all the people with the package already installed.  
For example, I have installed it, and already did the current migration. If you modify it instead of creating a new one, I cannot  have this index created by the package.

**Comment by atymic** [comment]
> Are you sure? I think the current migrations should not be touched, moreso in a package. Because you then break the migrations for all the people with the package already installed.
> For example, I have installed it, and already did the current migration. If you modify it instead of creating a new one, I cannot have this index created by the package.

Fair point. I just don't like having lots of package migrations clogging up the folder, but since we don't actually publish the migrations it's probably ok. 
My thinking was that this isn't actually required for the package.

thoughts @thomasjohnkane ?

**Comment by atymic** [comment]
Unless i'm missing something, I don't think this would actually speed up the queries. Since we never query only the `send_at` column, we need to do a table scan anyway?

**Comment by nunodonato** [comment]
I was digging issues looking for this. Had to create a manual migration to add the index to send_at. @atymic can you explain your reasoning? The `snooze:send` command will query the `send_at` and, for big databases, this index will matter.

**Pull request closed without merging**